### PR TITLE
kubectl: respect DaemonSet strategy parameters for rollout status

### PR DESCRIPTION
**Pull request by lukaszo**
It handles "after-merge" comments from #41116

cc @kargakis @janetkuo 

I will add one more e2e test later. I need to handle some in company stuff.

### CODE REVIEW

**Comment by lukaszo** [comment]
@k8s-bot bazel test this

File: pkg/kubectl/rollout_status.go [hunk 1]
```diff
@@ -98,8 +99,14 @@ func (s *DaemonSetStatusViewer) Status(namespace, name string, revision int64) (
 		if daemon.Status.UpdatedNumberScheduled < daemon.Status.DesiredNumberScheduled {
 			return fmt.Sprintf("Waiting for rollout to finish: %d out of %d new pod have been updated...\n", daemon.Status.UpdatedNumberScheduled, daemon.Status.DesiredNumberScheduled), false, nil
 		}
-		if daemon.Status.NumberAvailable < daemon.Status.DesiredNumberScheduled {
-			return fmt.Sprintf("Waiting for rollout to finish: %d of %d updated pods are available...\n", daemon.Status.NumberAvailable, daemon.Status.DesiredNumberScheduled), false, nil
+
+		var maxUnavailable int
+		if daemon.Spec.UpdateStrategy.Type == extensions.RollingUpdateDaemonSetStrategyType {
```
Comment (kargakis) [inline pkg/kubectl/rollout_status.go]: Does it make sense for us to support DS in `rollout status` if the RollingUpdate strategy is not used?

Comment (lukaszo) [inline pkg/kubectl/rollout_status.go]: I think yes, even if pods are deleted manually someone may want to see the status using kubectl.

**Comment by lukaszo** [comment]
@k8s-bot non-cri e2e test this

File: pkg/kubectl/rollout_status.go (hunk 1 above)
Comment (janetkuo) [inline pkg/kubectl/rollout_status.go]: > I think yes, even if pods are deleted manually someone may want to see the status using kubectl.

+1

**Comment by janetkuo** [comment]
LGTM

**Review by janetkuo — APPROVED** [review]

File: pkg/kubectl/rollout_status.go (hunk 1 above)
Comment (kargakis) [inline pkg/kubectl/rollout_status.go]: The thing is that OnDelete is not really a strategy but a placeholder for not having a strategy (iow. delegating to a custom process). This change enforces maxUnavailable=0 which is not necessarily the case with every custom strategy.

**Comment by kargakis** [comment]
It's better to disallow `rollout status` for OnDelete now, open an issue to discuss it, and possibly enable it in the future, rather than enable it now and realize in the future that it's not really what we wanted to support.

**Comment by janetkuo** [comment]
I'm fine with disabling this for `OnDelete` for now. It gives us more flexibility.

File: pkg/kubectl/rollout_status.go (hunk 1 above)
Comment (lukaszo) [inline pkg/kubectl/rollout_status.go]: done

**Comment by lukaszo** [comment]
I've updated code after the comments and added unit tests.

**Comment by kargakis** [comment]
@lukaszo you need to update bazel

LGTM

**Comment by lukaszo** [comment]
@kargakis it wasn't bazel, go vet returned two errors.

**Pull request merged**